src/main/java/org/damap/base/rest/storage/InternalStorageTranslationValidator.java

```diff
@@ -0,0 +1,90 @@
+      InternalStorageRepo internalStorageRepo,
+      InternalStorageTranslationRepo internalStorageTranslationRepo)
+      throws ClientErrorException {
+    log.info("Validating internal storage translation for creation");
+
+    // Check if no translation for the same language exists
+    if (internalStorageTranslationRepo.existsTranslationForStorageIdAndLanguageCode(
+        internalStorageTranslationDO.getStorageId(),
+        internalStorageTranslationDO.getLanguageCode())) {
+      throw new ClientErrorException(
+          "Translation for language code "
+              + internalStorageTranslationDO.getLanguageCode()
+              + " already exists",
+          Response.Status.BAD_REQUEST);
+    }
+
+    InternalStorageTranslationValidator.validationCommon(
+        internalStorageTranslationDO, internalStorageRepo);
+  }
+
+  public void validateForUpdate(
+      String id,
+      InternalStorageTranslationDO internalStorageTranslationDO,
+      InternalStorageTranslationRepo internalStorageTranslationRepo,
+      InternalStorageRepo internalStorageRepo)
+      throws ClientErrorException {
+    log.info("Validating internal storage translation for update");
+
+    InternalStorageTranslation internalStorageTranslation =
+        internalStorageTranslationRepo.findById(Long.parseLong(id));
+
+    if (internalStorageTranslation == null) {
+      throw new NotFoundException("No internal storage translation with ID " + id + " found");
+    }
+
+    if (internalStorageTranslationRepo.existsTranslationForStorageIdAndLanguageCodeExceptId(
+        internalStorageTranslationDO.getStorageId(),
+        internalStorageTranslationDO.getLanguageCode(),
+        Long.parseLong(id))) {
+      throw new ClientErrorException(
+          "Translation for language code "
+              + internalStorageTranslationDO.getLanguageCode()
+              + " already exists",
+          Response.Status.BAD_REQUEST);
+    }
+
+    InternalStorageTranslationValidator.validationCommon(
+        internalStorageTranslationDO, internalStorageRepo);
+  }
+
+  public void validationCommon(
+      InternalStorageTranslationDO internalStorageTranslationDO,
+      InternalStorageRepo internalStorageRepo)
+      throws ClientErrorException {
+    log.info("Validating common internal storage translation");
+
+    InternalStorage internalStorage =
+        internalStorageRepo.findById(internalStorageTranslationDO.getStorageId());
+
+    if (internalStorage == null) {
+      throw new NotFoundException(
+          "No internal storage found for id " + internalStorageTranslationDO.getStorageId());
+    }
+
+    // Check if language code is 'deu' or 'eng'
+    if (!internalStorageTranslationDO.getLanguageCode().equals("deu")
+        && !internalStorageTranslationDO.getLanguageCode().equals("eng")) {
+      throw new ClientErrorException(
+          "Language code must be 'deu' or 'eng'", Response.Status.BAD_REQUEST);
+    }
```

Comment:
minor/shelve: Might make sense to fetch available languages from the database and compare with them. Probably not really relevant right now.